### ISSUE TITLE
RN 0.72: Broken border colors / radius on Android

### ISSUE THREAD

**Issue by MicroDroid** [facebook/react-native#37753]
### Description

When `borderRadius` is set, `borderColor` is `backgroundColor` by default (which is wrong), and only `borderColor` can change its color, `borderTopColor` etc. don't work (also wrong).

### React Native Version

0.72.0-rc.5

### Output of `npx react-native info`

```
info Fetching system and libraries information...
System:
  OS: Linux 5.15 Arch Linux
  CPU: (16) x64 AMD Ryzen 7 5800H with Radeon Graphics
  Memory: 13.05 GB / 31.35 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.14.2
    path: ~/.nvm/versions/node/v18.14.2/bin/node
  Yarn:
    version: 3.5.0
    path: /usr/sbin/yarn
  npm:
    version: 9.5.0
    path: ~/.nvm/versions/node/v18.14.2/bin/npm
  Watchman:
    version: 20221016.020512.0
    path: /usr/sbin/watchman
SDKs:
  Android SDK:
    API Levels:
      - "31"
      - "33"
      - "33"
    Build Tools:
      - 30.0.2
      - 30.0.3
      - 31.0.0
      - 33.0.0
      - 33.0.2
    System Images:
      - android-33 | Google APIs Intel x86_64 Atom
    Android NDK: Not Found
IDEs:
  Android Studio: AI-222.4459.24.2221.9971841
Languages:
  Java:
    version: 11.0.17
    path: /usr/sbin/javac
  Ruby: Not Found
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: Not found
  ```

### Steps to reproduce

1. Use the example code I provided
2. Observe border color is green (same as background)
3. Change `borderTopColor` to `borderColor`
4. Observe border color becomes orange

### Snack, code example, screenshot, or link to a repository

```tsx
<View
	style={{
		width: 128,
		height: 128,
		borderWidth: 5,
		borderRadius: 5,
		borderTopColor: 'orange',
		backgroundColor: 'green',
	}}
/>
```

**Comment by cortinico** [comment]
@MicroDroid could you test on 0.71 and check if this is a regression introduced in 0.72?

**Comment by MicroDroid** [comment]
@cortinico this bug appeared after upgrading to 0.72 so presumably yeah

**Comment by MicroDroid** [comment]
(the app was on 0.71.8 iirc)

**Comment by NickGerleman** [comment]
Iirc there were some changes here added for paper impl of web props. Let me see if I can find the commit.

**Comment by NickGerleman** [comment]
Relevant changes:
1. https://github.com/facebook/react-native/pull/35572
2. https://github.com/facebook/react-native/pull/35999
3. https://github.com/facebook/react-native/pull/36238

cc @gabrieldonadel @lunaleaps @necolas

**Comment by kelset** [comment]
thanks for reporting! This looks like something that needs to be addressed ASAP

**Comment by MicroDroid** [comment]
(and I don't want to wait 2 more weeks for another RC)

**Comment by Pranav-yadav** [comment]
> So once you fix this, could I install `react-native` package from `main` GitHub branch so that I can make a new release for the app I'm working on? Or would that not work?
> 
> I ask because I am not sure how native code is bundled or whatever.

@MicroDroid Actually, `main` is NOT stable. The stable versions' code lives in their own `*-stable` branch (e.g.: `0.71-stable`) and the respective RCs, stable, and patches are released from these respective branches only.
Once the PR w/ fix is opened/merged you can expect more details about the same under this issue only :)

> (and I don't want to wait 2 more weeks for another RC)

Hopefully, that won't be the case, as Lorenzo said 👇:
>"This looks like something that needs to be addressed **_ASAP_**" - https://github.com/facebook/react-native/issues/37753#issuecomment-1582346453

**Comment by NickGerleman** [comment]
@gabrieldonadel is this something that you might be willing to take a look at as the author of the changes?

We can also otherwise revert these and see if it resolves the issue. Though due to the nature of the source build, it does seem like that might need another RC if we cannot locally repro.

**Comment by fabriziobertoglio1987** [comment]
I can start working on this task. I remain available. What timeframe do I have? Thanks

**Comment by fabriziobertoglio1987** [comment]
- Reproduced the issue in the main branch. 
- I'm reviewing the logic in ReactBackgroundViewDrawable [drawRoundedBackgroundWithBorders](https://github.com/fabriziobertoglio1987/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L355) and the diff from the prs

**Comment by fabriziobertoglio1987** [comment]
1) The first issue is probably connected to the example itself, we should use borderWidthTop instead of borderWidth. I don't think this is an issue as the check was introduced in 2018 with this [commit](https://github.com/fabriziobertoglio1987/react-native/commit/c5ca26a0e5c0660196300ee34d6007c63879611f#diff-fc179133a5281ac8df07a8d0ce0878b2beb1e0725076232b0b07541cc0e0b369R434-R445).


```xml
<View
  style={{
    width: 128,
    height: 128,
    borderWidthTop: 5,
    borderRadius: 5,
    borderTopColor: 'orange',
    backgroundColor: 'green',
  }}
/>
```

2) The second issue, for which I will publish a PR. colorTop seems to use the default of black instead of the color set from javascript 

```java
if (borderWidth.top > 0) {
  final float x1 = left - mGapBetweenPaths;
  final float y1 = top;
  final float x2 = mInnerTopLeftCorner.x - mGapBetweenPaths;
  final float y2 = mInnerTopLeftCorner.y;
  final float x3 = mInnerTopRightCorner.x + mGapBetweenPaths;
  final float y3 = mInnerTopRightCorner.y;
  final float x4 = right + mGapBetweenPaths;
  final float y4 = top;

  // drawQuadrilateral(canvas, Color.BLUE, x1, y1, x2, y2, x3, y3, x4, y4);
  drawQuadrilateral(canvas, colorTop, x1, y1, x2, y2, x3, y3, x4, y4);
}
```

| Before    | After |
| ----------- | ----------- |
| <image src="https://github.com/facebook/react-native/assets/24992535/2fbec962-402b-45d9-8c1b-ec3b2e0ebb20" width="350" />      | <image src="https://github.com/facebook/react-native/assets/24992535/15bf707d-b1bb-4096-9676-8171e8873b12" width="350" />       |

**Comment by MicroDroid** [comment]
@fabriziobertoglio1987 I don't know if the example I put up is accurate, but I expect my UI elements to not lose borders due to a RN upgrade, is all

**Comment by fabriziobertoglio1987** [comment]
<details><summary>Individual border colors not working</summary>
<p>

Using borderColorTop instead of borderColor ([commit](https://github.com/fabriziobertoglio1987/react-native/commit/79853d6e53337c82f064aa80dd8b8ac8916afa9d))

```javascript
const styles = StyleSheet.create({
  view: {
    width: 128,
    height: 128,
    borderTopWidth: 5,
    borderRadius: 5,
    borderColor: 'orange',
    backgroundColor: 'green',
  },
  button: {},
});

function TextExample() {
  const [redColor, setRedColor] = React.useState(true);
  return (
    <>
      <View
        style={[
          styles.view,
          // works with borderColor 
          {borderTopColor: redColor ? '#ff0000' : '#3399ff'},
        ]}
      />
      <Button title="change color" onPress={() => setRedColor(prev => !prev)} />
      <Text>
        Current color is {redColor ? '#ff0000' : '#3399ff'} which corresponds to{' '}
        {redColor ? 'red color' : 'blue color'}
      </Text>
    </>
  );
}
```

| borderColor    | borderTopColor |
| ----------- | ----------- |
| <image src="https://user-images.githubusercontent.com/24992535/244871274-9375578e-d470-43e1-9f56-4aee25ba9c2c.png" width="350" />      | <image src="https://user-images.githubusercontent.com/24992535/244871514-028c6919-9ae2-4094-afb9-4a55022e3b29.png" width="350" />       |

</p>
</details>

**Comment by fabriziobertoglio1987** [comment]
<details><summary>Border radius not working</summary>
<p>

<img width="1920" alt="Screenshot 2023-06-10 at 8 54 42 PM" src="https://github.com/facebook/react-native/assets/24992535/ec139800-0fdb-4091-98ee-aaa349e5fc32">


<img width="1920" alt="Screenshot 2023-06-10 at 8 55 37 PM" src="https://github.com/facebook/react-native/assets/24992535/0bffaaf6-2565-4aec-b102-c18a774d501e">


</p>
</details>

**Comment by fabriziobertoglio1987** [comment]
The issue is caused by the missing clipping effect of the top border when using borderColorTop. 
The same issue does not reproduce when using borderColor.

This is the result after commenting the logic responsible for [drawing top border](https://github.com/facebook/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L472). 

- When using borderColor, the rectangle is clipped on the top to display the border
- When using borderTopColor, the rectangle is not clipped on the top and the border is not displayed

| borderColor (fixed)    | borderTopColor (not fixed) |
| ----------- | ----------- |
| <image src="https://user-images.githubusercontent.com/24992535/244877294-b2f93151-91b1-46f1-bbdd-6f0d711755bc.png" width="350" />      | <image src="https://user-images.githubusercontent.com/24992535/244877399-b17a2392-b496-4fcc-9764-845bc03062ab.png" width="350" />       |

**Comment by fabriziobertoglio1987** [comment]
Borders are drawn by using an inner and outer rectangle responsible for clipping the original view.
- The inner rectangle clips the view and draws the border inside it
- The outer rectangle clips the view and applies the borderRadius

STEP 1 - Drawing inner rectangle with rounded borders (1 for each side when using individual colors with borderTopColor).

- Calculating the outer rectangle with rounded borders for each size (top, left, bottom, right) ([sourcecode](https://github.com/facebook/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L879-L903
))
- Clipping the View to draw the border ([sourcecode](https://github.com/facebook/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L472))

After: Changed logic to `drawQuadrilateral(canvas, -65536, -0.8f, 0.0f, -0.8f, 27.5f, 352.8f, 27.5f, 352.8f, 0.0f);`

| Before    | After |
| ----------- | ----------- |
| <image src="https://github.com/facebook/react-native/assets/24992535/8dde0c2d-5fda-4d9b-9ce8-b4908bfa30c8" width="350" />      | <image src="https://github.com/facebook/react-native/assets/24992535/2c5d6a60-9919-46d0-95c2-d4b7dc906d39" width="350" />       |

STEP 2 - The borderRadius is applied by clipping the View with an outside rectangle with rounded corners. 

https://github.com/facebook/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L343

| Before    | After |
| ----------- | ----------- |
| <image src="https://github.com/facebook/react-native/assets/24992535/4696dc61-9663-40b9-b153-3d5e36b7f268" width="350" />      | <image src="https://github.com/facebook/react-native/assets/24992535/9f1b7677-facd-45f6-a35c-ec68fe641e37" width="350" />       |

**Comment by fabriziobertoglio1987** [comment]
STEP 1) Using hardcoded logic to draw inner rectangle  

```
drawQuadrilateral(canvas, -65536, -0.8f, 0.0f, -0.8f, 27.5f, 352.8f, 27.5f, 352.8f, 0.0f);
```

STEP 2) Using borderTopWidth instead of borderWidth (separate issue with [setBorderWidth](https://github.com/fabriziobertoglio1987/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L207))

```javascript
const styles = StyleSheet.create({
  view: {
    width: 128,
    height: 128,
    borderTopWidth: 5,
    borderRadius: 5,
    backgroundColor: 'green',
  },
  button: {},
});

function TextExample() {
  const [redColor, setRedColor] = React.useState(true);
  return (
    <>
      <View
        style={[
          styles.view,
          {borderTopColor: redColor ? '#ff0000' : '#3399ff'},
        ]}
      />
      <Button title="change color" onPress={() => setRedColor(prev => !prev)} />
      <Text>
        Current color is {redColor ? '#ff0000' : '#3399ff'} which corresponds to{' '}
        {redColor ? 'red color' : 'blue color'}
      </Text>
    </>
  );
}
```

| Before    | After |
| ----------- | ----------- |
| <image src="https://github.com/facebook/react-native/assets/24992535/6f1280f6-9a68-4ee8-8aed-ea20584a4ba2" width="350" />      | <image src="https://github.com/facebook/react-native/assets/24992535/ada65fc0-6023-4a44-a3d8-baf752041b48" width="350" />       |

**Comment by fabriziobertoglio1987** [comment]
The top border is drawn correctly when using borderTopWidth and borderColor, which confirm issue with the borderTopColor implementation in Java as explained above.

```javascript
const styles = StyleSheet.create({
  view: {
    width: 128,
    height: 128,
    borderTopWidth: 5,
    borderRadius: 5,
    backgroundColor: 'green',
  },
  button: {},
});

function TextExample() {
  const [redColor, setRedColor] = React.useState(true);
  return (
    <>
      <View
        style={[styles.view, {borderColor: redColor ? '#ff0000' : '#3399ff'}]}
      />
      <Button title="change color" onPress={() => setRedColor(prev => !prev)} />
      <Text>
        Current color is {redColor ? '#ff0000' : '#3399ff'} which corresponds to{' '}
        {redColor ? 'red color' : 'blue color'}
      </Text>
    </>
  );
}
```

<image src="https://github.com/facebook/react-native/assets/24992535/1d61a6ae-ea92-4ba3-88fc-beb43f6ab5c4" width="250" />

**Comment by fabriziobertoglio1987** [comment]
The functionality works on Expo

<img width="1525" alt="Screenshot 2023-06-11 at 8 32 43 PM" src="https://github.com/facebook/react-native/assets/24992535/c9d98f5d-9e05-476c-abb8-bcc30565842f">


The functionality does not work on 0.71.8, adding a black full border (borderColor: "black") will remove the top border (borderTopColor: "red")

<img width="1920" alt="Screenshot 2023-06-11 at 8 07 27 PM" src="https://github.com/facebook/react-native/assets/24992535/60a84a53-239b-4385-8b5e-59e5818c32da">

**Comment by fabriziobertoglio1987** [comment]
STEP 2 - Clipping inner rectangle (borders)

https://github.com/facebook/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L402


<details><summary>CLICK TO OPEN TESTS RESULTS</summary>
<p>



```javascript
const styles = StyleSheet.create({
  view: {
    width: 128,
    height: 128,
    borderTopWidth: 5,
    borderLeftWidth: 5,
    borderRightWidth: 5,
    borderRadius: 5,
    borderTopColor: 'orange',
    backgroundColor: 'green',
  },
});
```
| Before   | After |
| ----------- | ----------- |
| <image src="https://user-images.githubusercontent.com/24992535/244940457-40058980-8913-41a6-b39c-56c550fa0eb2.png" width="350" />      | <image src="https://user-images.githubusercontent.com/24992535/244941341-9bb2cf6b-9919-4d28-b740-0ac0cfc111b6.png" width="350" />       |

### Functionality does not work on 4 borders

```javascript
const styles = StyleSheet.create({
  view: {
    width: 128,
    height: 128,
    borderTopWidth: 5,
    borderLeftWidth: 5,
    borderRightWidth: 5,
    borderBottomWidth: 5,
    borderRadius: 5,
    borderTopColor: 'orange',
    backgroundColor: 'green',
  },
});
```


| Before    | After |
| ----------- | ----------- |
| <image src="https://user-images.githubusercontent.com/24992535/244940464-59825e98-f752-40f1-9d6c-669eef759fb8.png" width="350" />      | <image src="https://user-images.githubusercontent.com/24992535/244941298-44311c2c-6d80-4f95-800b-adad8601ebfc.png" width="350" />       |tangle (borders)

</p>
</details>

**Comment by fabriziobertoglio1987** [comment]
Borders are drawn by using an inner and outer rectangle responsible for clipping the original view.
- The inner rectangle clips the view and draws the border inside it
- The outer rectangle clips the borders inside the view

STEP 1 - The borderRadius is applied by clipping the View with an outside rectangle with rounded corners. 

https://github.com/facebook/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L343

| Before    | After |
| ----------- | ----------- |
| <image src="https://github.com/facebook/react-native/assets/24992535/4696dc61-9663-40b9-b153-3d5e36b7f268" width="350" />      | <image src="https://github.com/facebook/react-native/assets/24992535/9f1b7677-facd-45f6-a35c-ec68fe641e37" width="350" />       |

STEP 2 - Clipping inner rectangle (borders) 

https://github.com/facebook/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L402


<details><summary>Difference when clipping view with 3 borders instead of 4 borders</summary>
<p>

### Clipping a View with 3 borders

```javascript
const styles = StyleSheet.create({
  view: {
    width: 128,
    height: 128,
    borderTopWidth: 5,
    borderLeftWidth: 5,
    borderRightWidth: 5,
    borderRadius: 5,
    borderTopColor: 'orange',
    backgroundColor: 'green',
  },
});
```
| Before   | After |
| ----------- | ----------- |
| <image src="https://user-images.githubusercontent.com/24992535/244941341-9bb2cf6b-9919-4d28-b740-0ac0cfc111b6.png" width="350" />      | <image src="https://user-images.githubusercontent.com/24992535/244940457-40058980-8913-41a6-b39c-56c550fa0eb2.png" width="350" />       |

### Clipping a View with 4 borders

```javascript
const styles = StyleSheet.create({
  view: {
    width: 128,
    height: 128,
    borderTopWidth: 5,
    borderLeftWidth: 5,
    borderRightWidth: 5,
    borderBottomWidth: 5,
    borderRadius: 5,
    borderTopColor: 'orange',
    backgroundColor: 'green',
  },
});
```


| Before    | After |
| ----------- | ----------- |
| <image src="https://user-images.githubusercontent.com/24992535/244941298-44311c2c-6d80-4f95-800b-adad8601ebfc.png" width="350" />      | <image src="https://user-images.githubusercontent.com/24992535/244940464-59825e98-f752-40f1-9d6c-669eef759fb8.png" width="350" />       |tangle (borders)

</p>
</details>

STEP 3 - Drawing inner rectangle with rounded borders (1 for each side when using individual colors with borderTopColor).

- calculating coordinates (top, left, bottom, right) ([sourcecode](https://github.com/facebook/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L879-L903
))
- drawing borders ([sourcecode](https://github.com/facebook/react-native/blob/03f70bf995379f08a77abcf96bb0e31ff75ca8c3/packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/ReactViewBackgroundDrawable.java#L472))

| Before    | After |
| ----------- | ----------- |
| <image src="https://github.com/facebook/react-native/assets/24992535/8dde0c2d-5fda-4d9b-9ce8-b4908bfa30c8" width="350" />      | <image src="https://github.com/facebook/react-native/assets/24992535/2c5d6a60-9919-46d0-95c2-d4b7dc906d39" width="350" />       |

Conclusions: 
- The black borders are drawn on top of the red border

**Comment by kelset** [comment]
Thanks @fabriziobertoglio1987 for the investigation!

I've spent some time this morning diving a bit more into this from the perspective of trying to figure out which commit created the problem; here are a few things I've learned.

First off, to repro this problem is also possible to quickly just add the sample code in RNTester, under `packages/rn-tester/js/examples/View/ViewExample.js`; here's how things look in the 0.72-stable branch with the test code added:
<img width="1502" alt="Screenshot 2023-06-12 at 11 23 21" src="https://github.com/facebook/react-native/assets/16104054/4edc22b1-248a-48f9-9318-6fd5449292f2">

So, yes, the Android issue is fully confirmed there. After setting that up, made a local branch and tried to revert the commits that @NickGerleman pointed out.

By reverting "feat: Add logical border block color properties" https://github.com/facebook/react-native/commit/597a1ff60b3e1844b4794fb4acd40fa073f2e93b I actually got some improvement to the situation, shown here:
<img width="824" alt="revert just logical border block" src="https://github.com/facebook/react-native/assets/16104054/2f31f8ce-8e27-44fb-ac83-014fb6b2f841">

Reverting "feat: Add logical border radius implementation" https://github.com/facebook/react-native/commit/4ae4984094e4846bc2bc0e3374ab5d934ee6bc5f on top of the other revert didn't change anything:
<img width="844" alt="revert both" src="https://github.com/facebook/react-native/assets/16104054/7dd38fb2-fff9-448b-994f-c68bdf7cda16">

And this is consistent with what Fabrizio is saying, meaning that there was a second underlying bug already present since 0.71.x. I didn't dig into what in 0.71 might have caused the problem.

So this leaves us into a position where:
* there's a bug with Android and borders that was already existing in 0.71, that should be handled
* there's a bigger regression on top of it, the one reported here, that is caused by https://github.com/facebook/react-native/commit/597a1ff60b3e1844b4794fb4acd40fa073f2e93b by @gabrieldonadel 

I think that if viable, we should properly fix both.

If that's too complicated/would take more than a couple days, we should revert the commit by Gabriel, bring back the status of things to the one we had in 0.71, and proceed with 0.72.0 while proper fixes are landed in main and then backported to 0.71 and 0.72.

**Comment by gabrieldonadel** [comment]
> * there's a bigger regression on top of it, the one reported here, that is caused by [597a1ff](https://github.com/facebook/react-native/commit/597a1ff60b3e1844b4794fb4acd40fa073f2e93b) by @gabrieldonadel

This is fixed here https://github.com/facebook/react-native/pull/37828

**Comment by cipolleschi** [comment]
For sake of completeness, @cortinico found this commit which touched that file in 0.72
- https://github.com/gabrieldonadel/react-native/commit/1d5103227851ab92de889d5e7e910393b5d8743a
It does not seems related to the issue to me, tbh, but l'm putting it there to keep track of it.

**Comment by gabrieldonadel** [comment]
Ok, I did some more investigation on this. I tested multiple versions of react-native until I found the one where `borderColor` + `borderRadius` was working, the last version where this was working 100% was `0.70.6`, which means that something on 0.71 broke this 

<img width="482" alt="image" src="https://github.com/facebook/react-native/assets/11707729/9b6775eb-a1d2-414d-b7d1-1be1954b139b">

**Comment by lunaleaps** [comment]
Thanks @gabrieldonadel for your amazing help!

We'll pick the following into 0.72
* Revert of https://github.com/facebook/react-native/commit/cd6a91343ee24af83c7437b3f2449b41e97760e9 - reverting this fixes ignored side-specific borderColor when borderRadius set
* https://github.com/facebook/react-native/commit/7d1f7f3f5fbc9e659757f17fae98fa8778ffdbb6 - fix for the default borderColor matching the backgroundColor when borderRadius set.

**Comment by leymytel** [comment]
Hello everyone.

This issue seems to have similarities with a previously reported issue in version 0.71.x.

Here's the link to the previous issue for reference: [#36569](https://github.com/facebook/react-native/issues/36569).

**Comment by kelset** [comment]
great stuff @gabrieldonadel! I locally applied both of your PRs on top of 0.72 and I can confirm that with both, the regression is fully addressed 🎉

<img width="811" alt="Screenshot 2023-06-13 at 10 52 58" src="https://github.com/facebook/react-native/assets/16104054/39acb7d8-78bf-4947-9e8c-6ce8adb0cb05">

**Comment by kelset** [comment]
we've done the cherry picking, we're starting testing, unless something unexpected goes horribly wrong in the next 12/24hrs RC6 should come out and soon after 0.71.11

**Comment by kelset** [comment]
update, released both:
* https://github.com/facebook/react-native/releases/tag/v0.71.11
* https://github.com/facebook/react-native/releases/tag/v0.72.0-rc.6
